### PR TITLE
Friendship

### LINKED ISSUE #11: Dashboard: Friends
As an authenticated user,
I see a section for friends,
In this section, there should be a text field to enter a friend's email and a button to "Add Friend"

**Scenarios**: 
1. If I have not added any friends there should be a message. "You currently have no friends".

2. If I have added friends, I should see a list of all my friends.

**Details:** Users should be able to add a friend by their email address, as long as, the friend is a user of our application and exists in our database. 

**Example:** 
Bugs Bunny and Lola Bunny are users of our application, but Daffy Duck is not. 
1. When Bugs Bunny enters `lola_bunny@gmail.com` to add friend it will be successful and Lola should show up as Bugs Bunny's friend.
2. When Bugs Bunny enters `daffy_duck@gmail.com` to add friend it should give an error message that the user does not exist. 

**Tips:** You'll want to research self-referential `has_many through`. [Here](http://blog.hasmanythrough.com/2007/10/30/self-referential-has-many-through) is a good starting point to understand the concept. You will probably need to do more googling but that's part of the fun ;)

- [x] Write a happy path test 
- [x] Write a sad path test

### PR DESCRIPTION
**Pull request by coreycartercodes**
To-Do:
- Create Specs
- Fix reciprocity when adding friends
***Hold off on merging until this is done***

### CODE REVIEW

**Comment by coreycartercodes** [comment]
Fixed reciprocity issues through manual adding.

**Comment by coreycartercodes** [comment]
Added specs for users and friendships.

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,9 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+
+  def generate_flash(resource)
+    resource.errors.messages.each do |validation, message|
+      flash[validation] = "#{validation}: #{message}"
+    end
+  end
```
Comment (lkriffell) [inline app/controllers/application_controller.rb:8]: I really like what you did here. Never would have thought about doing this

File: db/seeds.rb [hunk 2]
```diff
@@ -5,3 +5,6 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
+
+    friend_1 = User.create(email: "a@b.com", password: "password")
+    friend_2 = User.create(email: "b@c.com", password: "password")
```
Comment (lkriffell) [inline db/seeds.rb:10]: Just fyi you can use factory bot within the seeds file too!

**Pull request merged**